packages/tests/server/react/ensureQueryData.test.tsx

```diff
@@ -0,0 +1,75 @@
+import { createQueryClient } from '../__queryClient';
+import { createAppRouter } from './__testHelpers';
+import { QueryClientProvider, useQueryClient } from '@tanstack/react-query';
+import { render, waitFor } from '@testing-library/react';
+import React, { useEffect, useState } from 'react';
+
+let factory: ReturnType<typeof createAppRouter>;
+beforeEach(() => {
+  factory = createAppRouter();
+});
+afterEach(async () => {
+  await factory.close();
+});
```

Comment:
We usually use `konn` for tests

Reply:
From what I see, there are still plenty of tests using the factory, so I reused the same style here.

```
- setQueryData
- setInfiniteQueryData
- prefetchQuery
- infiniteQuery
- invalidateQueries
```

They all use it, and because `ensureQueryData` is about the same part of the API, I made the test the same as well. Didn't want it to diverge.

Reply:
Ah ok :) great then